solidity/src/Base7683.sol

```diff
@@ -3,82 +3,123 @@ pragma solidity 0.8.25;
 import { IPermit2, ISignatureTransfer } from "@uniswap/permit2/src/interfaces/IPermit2.sol";
 
 import {
     GaslessCrossChainOrder,
     OnchainCrossChainOrder,
     ResolvedCrossChainOrder,
-    Output,
-    FillInstruction,
     IOriginSettler,
     IDestinationSettler
 } from "./ERC7683/IERC7683.sol";
-import { OrderData, OrderEncoder } from "./libs/OrderEncoder.sol";
 
+/**
+ * @title Base7683
+ * @notice Implements the ERC7683 standard for cross-chain order resolution, filling, settlement, and refunding.
+ * @author BootNode
+ * @dev Contains logic for managing orders without requiring specifics of the order data type.
+ * Notice that settling and refunding is not described in the ERC7683 but it is included here to provide a common
+ * interface for solvers to use.
+ */
 abstract contract Base7683 is IOriginSettler, IDestinationSettler {
     // ============ Libraries ============
     using SafeERC20 for IERC20;
 
     // ============ Constants ============
-
-    enum OrderStatus {
-        UNFILLED,
-        OPENED,
-        FILLED,
-        SETTLED,
-        REFUNDED
-    }
-
+    /// @notice The instance of the Permit2 contract.
     IPermit2 public immutable PERMIT2;
 
-    bytes32 public constant GASLESS_CROSS_CHAIN_ORDER_TYPEHASH = keccak256(
-        "GaslessCrossChainOrder(address originSettler,address user,uint256 nonce,uint64 originChainId,uint32 openDeadline,uint32 fillDeadline,bytes32 orderDataType,bytes orderData)"
+    /// @notice Type hash used for encoding ResolvedCrossChainOrder.
+    bytes32 public constant RESOLVED_CROSS_CHAIN_ORDER_TYPEHASH = keccak256(
+        "ResolvedCrossChainOrder(address user, uint64 originChainId, uint32 openDeadline, uint32 fillDeadline, Output[] maxSpent, Output[] minReceived, FillInstruction[] fillInstructions)Output(bytes32 token, uint256 amount, bytes32 recipient, uint64 chainId)FillInstruction(uint64 destinationChainId, bytes32 destinationSettler, bytes originData)"
     );
 
+    /// @notice The witness type string used in PERMIT2 transactions.
     string public constant witnessTypeString =
-        "GaslessCrossChainOrder witness)GaslessCrossChainOrder(address originSettler,address user,uint256 nonce,uint64 originChainId,uint32 openDeadline,uint32 fillDeadline,bytes32 orderDataType,bytes orderData)TokenPermissions(address token,uint256 amount)";
+        "ResolvedCrossChainOrder witness)ResolvedCrossChainOrder(address user, uint64 originChainId, uint32 openDeadline, uint32 fillDeadline, Output[] maxSpent, Output[] minReceived, FillInstruction[] fillInstructions)Output(bytes32 token, uint256 amount, bytes32 recipient, uint64 chainId)FillInstruction(uint64 destinationChainId, bytes32 destinationSettler, bytes originData)TokenPermissions(address token,uint256 amount)";
+
+    /// @notice Possible statuses for an order. Other possible statuses should be defined in the inheriting contract.
+    bytes32 public constant UNKNOWN = "";
```

Comment:
I'm curious why you moved this away from a enum to this?

Reply:
This is IMO confusing, as a consumer of these contracts, I want to be able to easily tell the set of valid values. I'd rather err on the side of simplicity and ergonomics over generalization here and just have Base7683 specify an enum with valid values, and if folks need more, they can add to the base.